manifests/charts/gateway/templates/_helpers.tpl

```diff
@@ -15,21 +15,19 @@ Create chart name and version as used by the chart label.
 
 {{- define "gateway.labels" -}}
 helm.sh/chart: {{ include "gateway.chart" . }}
-{{ include "gateway.selectorLabels" . }}
-{{- if .Chart.AppVersion }}
-app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
-{{- end }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
-app.kubernetes.io/name: {{ include "gateway.name" . }}
-{{- range $key, $val := .Values.labels }}
-{{- if not (or (eq $key "app") (eq $key "istio")) }}
-{{ $key | quote }}: {{ $val | quote }}
-{{- end }}
-{{- end }}
+{{ include "gateway.podLabels" . }}
```

Comment:
I am quite confused by this refactoring. Why would the gateway labels include pod labels ? Can we keep the code clean - even if it means some duplication of lines, there is no reason for all other resources to have the injection labels ( labels are indexed in k8s, so more expensive - and also create noise ). It's already pretty messy due to the 'reuse' and lack of clear spec on what labels we support for each object type - selectorLabels is also specific to pods, no need to put them on service or other objects.

My concern here is that we'll accidentally add even more labels or break backwards compat, and the code is already hard to read/review. If is intended for all resources - should be in gateway.labels, if it is pod specific - in podLabels or directly on the pod.

Reply:
please see https://github.com/istio/istio/pull/50540/files#r1582266135

- `master` currently has 
```
gateway.labels
  - gateway.selectorLabels

gateway.podLabels
  - gateway.selectorLabels

gateway.selectorLabels
```

my initial idea was to reuse the common labels between `gateway.labels` and `gateway.podLabels` by moving them to `gateway.podLabels`
```
gateway.labels
  - gateway.podLabels

gateway.podLabels
  - gateway.selectorLabels

gateway.selectorLabels
```

The problem with that and the current `gateway.podLabels` is that it represents "gateway.commonLabels"

Please see https://github.com/istio/istio/pull/50540/files#r1582266135 which has 
```
gateway.labels
  - gateway.selectorLabels

gateway.selectorLabels

gateway.sidecarInjectionLabels
```